### DISCUSSION TITLE
Customizable behavior for nearlyEqual: relative or absolute difference

### DISCUSSION THREAD

**Discussion by chidiwilliams** [josdejong/mathjs#2752] [Ideas]
_package.json_
```json
"mathjs": "^5.0.1"
````

I'm trying to set the epsilon value for comparing equality of two numbers. I want all numbers within a range of 1e-5 and below to be considered equal.

I'm doing:
```js
const math = require('mathjs');

math.config({
  epsilon: 1e-5,
});

console.log(math.equal(5e-10, 0));

```

But this gives an output of:
```js
false
```

I've also tried:
```js
const math = require('mathjs');
const mth = math.create({
  epsilon: 1e-5,
});

console.log(mth.equal(5e-10, 0));
```

But it prints the same output.

What's going on or what am I doing wrong?

**Comment by josdejong** [comment]
The `equal` function calculates the relative difference, not the absolute difference:

http://mathjs.org/docs/datatypes/numbers.html#equality

**Reply by gwhitney** [reply]
Actually, equal() is (currently) an undocumented mix of relative and absolute approximate comparisons. See #2838 for some proposals on how this might be systematized, and therefore cover your use case.

**Comment by chidiwilliams** [comment]
Ah, I see. I would love to see a method to compare absolute difference with EPSILON. Thanks anyway.

**Comment by josdejong** [comment]
Yes makes sense. 

As a workaround you could implement an equal function which does do two checks: one for the `value` itself, and one for `value+1`, something like:

```js
math.import({
  absEqual: (a, b) => math.equal(a, b) || math.equal(math.add(a, 1), math.add(b, 1))
})

math.config({epsilon: 1e-5})
console.log(math.equal(0, 1e-6))    // false
console.log(math.absEqual(0, 1e-6)) // true
```

If we implement absolute nearly equal functionality I think it should go in a new config option. It makes sense, let's give this some more thought. I will reopen this issue and change the title accordingly.

**Comment by gwhitney** [comment]
At this point, equality has both a relative and an absolute epsilon that it uses a mixture of; see https://mathjs.org/docs/core/configuration.html. You could set one to 0 to _only_ use a relative or an absolute tolerance. Hence, closing this discussion.
